crypto/store/store_init.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2001-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include "internal/store.h"
+#include "store_local.h"
+
+static CRYPTO_ONCE store_init = CRYPTO_ONCE_STATIC_INIT;
+DEFINE_RUN_ONCE_STATIC(do_store_init)
+{
+    return OPENSSL_init_crypto(0, NULL)
```

Comment:
Well, all I'm saying is I can't think of a reason for it (not that there isn't a lurking bug that makes it required). Consider this: libcrypto is supposed to be auto-initialising. If the store module only ever makes public API calls into other parts of libcrypto then it should auto-initialise for the store module too. The only ways to come unstuck are 1) if there is a bug that means it doesn't auto-init when it is supposed to or 2) if we are using some internal API to other parts of libcrypto that somehow avoid the auto-init. I suggest removing it, until we see a need for it.